contracts/linear/src/validator_pool.rs

```diff
@@ -340,12 +341,11 @@ impl LiquidStakingContract {
         self.add_whitelisted_validator(&validator_id, weight);
     }
 
-    pub fn add_validators(&mut self, validator_ids: Vec<AccountId>, weights: Vec<u16>) {
+    pub fn add_validators(&mut self, validator_id_to_weight: HashMap<AccountId, u16>) {
```

Comment:
Since both `Vec` and `HashMap` work well here, I don't feel a strong need of updating the function definition with `HashMap`. 

I'd rather make less change to keep the interface consistent with others, e.g. `update_base_stake_amounts`.

Reply:
Similarly, I would prefer define `update_weights` with a consistent interface.

Reply:
A parameter with a precise type can reduce the mental burden of developers. Since the range of `(vec[_], vec[_])` is much larger than `map<_, _>`, you can't avoid peoples from calling it using vectors with different lengths so you have to handle this case(costing more gas). Callers also have to make extra type transformations before calling it. Now that `add_validators` is not an interface directly faces end users, I think it worth a change.

Reply:
That's right. 

But the benefits of making the change is not quite tangible at this moment:
1. The current function parameter definition are not error prone and just work as expected.
2. The gas saving is very limited.

For a DeFi project, I'm not a fan of making changes to interfaces unless it's really necessary and beneficial, so I tend to make less change to interfaces, and keep consistent definition across validator related interfaces.

Reply:
```suggestion
    pub fn add_validators(&mut self, weights: HashMap<AccountId, u16>) {
```